float8_experimental/float8_dynamic_linear.py

```diff
@@ -62,8 +62,8 @@ class Float8DynamicLinear(torch.nn.Linear):
     def __init__(self, **super_kwargs):
         super().__init__(**super_kwargs)
 
-    def forward(self, x):
-        x_fp8 = cast_to_float8_e4m3fn(x, self.forward_config)
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
```

Comment:
nit: the `x`, `w`, `dL_dY` terminology is used throughout this repo, would be good to either stay consistent to that or if we are changing it add a TODO to change it everywhere

Reply:
I changed from x to input because my linter was complaining that this signature doesn't follow the Nn.Linear subclassing

I can change it back I think we shud todo to switch over the forwards to use the input as a name

Reply:
actuallly I am going to change it everywhere

Reply:
I don't really like `input`, `weight`, `grad_output`, it's a lot of characters and will make the codebase less readable given how they are used.  If we are changing everywhere, I think small # of chars specifically for this is important.  I'd rather override the linter here and not match the `input` name.

Reply:
Hmm as you have already gotten a sense for I am very much in favor of more characters rather than less lol

that being said I think this is an actual issue though:

``` Python
import torch

import torch.nn as nn
import torch.nn.functional as F

from float8_experimental.float8_linear import Float8Linear
from float8_experimental.float8_linear_utils import swap_linear_with_float8_linear


class FeedForward(nn.Module):
    def __init__(self) -> None:
        super().__init__()
        self.w1 = nn.Linear(4096, 14336, bias=False)
        self.w3 = nn.Linear(4096, 14336, bias=False)
        self.w2 = nn.Linear(14336, 4096, bias=False)

    def forward(self, x: torch.Tensor) -> torch.Tensor:
        out_1 = self.w1(input=x)
        out_2 = self.w3(input=x)
        out_3 = self.w2(input=out_1 * out_2)
        return out_3


batch_size, num_tokens, embedding_dim = 1, 1, 4096
tst_input = torch.randn(batch_size, num_tokens, embedding_dim)

my_mod = FeedForward()

out_1 = my_mod(tst_input)

swap_linear_with_float8_linear(my_mod, Float8Linear, emulate=True)
my_mod(tst_input)
```

will raise with 
```Python
TypeError: Float8Linear.forward() got an unexpected keyword argument 'input'
```

We dont have control how users call their nn.Linear modules

Reply:
gotcha, that trumps what I said, makes sense